test/cmocka/src/math/arithmetic/exponential.c

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2022 Intel Corporation. All rights reserved.
+//
+// Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <math.h>
+#include <cmocka.h>
+#include <string.h>
+
+#include <sof/math/exp_fcn.h>
+#include <sof/audio/format.h>
+#include <rtos/string.h>
+#include <sof/common.h>
+
+#define ULP_TOLERANCE 5.60032793
+#define ULP_SCALE 0.0000999999999749
+/* testvector in Q4.28, -5 to +5 */
+	static const int32_t iv[128] = {
+	    -1342177280, -1321040630, -1299903980, -1278767330, -1257630680, -1236494030,
```

Comment:
Not loving the practice of testing only on a tiny handful of known inputs.  It would be no more difficult do generate random 32 bit input (rejecting the values outside the legal input range) and feed it to the function, then test against libc floating point output on the same (converted) values.  Then you could get coverage into the millions trivially.  (I'm thinking specifically about that unterminated loop and how it might misbehave only only a few input values)

Reply:
What extra problem you have regarding test coverage is unclear to me.
I believe this addresses your worry   [https://github.com/thesofproject/sof/pull/6655#discussion_r1042911188](url)
At the  same time, floating lib is in place to check the limit !!

Reply:
Don't "resolve" the conversion when the concern of the reviewer is "unclear to you". In this project, resolving all conversations is not required for merging

Reply:
Okay, then what exactly qualifies as agreement or disagreement?

In addition, I would be happy to respond if a useful example were to be given.

Reply:
Make the test with random values in the range or if keep this approach put this to a separate header file so that the HiFi version does not need to duplicate the data. Actually same test for both should do?